bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookkeeperInternalCallbacks.java

```diff
@@ -107,6 +107,13 @@
         void startTLSComplete(int rc, Object ctx);
     }
 
+    /**
+     * A callback interface for GetListOfEntriesOfALedger command.
+     */
+    public interface GetListOfEntriesOfALedgerCallback {
+        void getListOfEntriesOfALedgerComplete(int rc, long ledgerId, ByteBuf buffer, Object ctx);
```

Comment:
I would suggest trying to use `CompletableFuture` on newly introduced features.

Reply:
introducing CompletableFuture